src/sjcl-custom/sjcl-ecdsa-signdeterministic.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+let sjcl = require('sjcl');
+/* eslint-disable new-cap */
+
+sjcl.ecc.ecdsa.secretKey.prototype.generateK =
+function(hash, hashObject) {
+
+  let curve = this._curve;
+  let qlen = sjcl.bitArray.bitLength(curve.r.toBits());
+
+  /* Utility functions */
+  /* used to generate k and v */
+  function repeat(str, times) {
+    return (new Array(times + 1)).join(str);
+  }
+  function bits2int(bits) {
+
+    let blen = sjcl.bitArray.bitLength(bits);
+
+    if (blen > qlen) {
+      return sjcl.bn.fromBits(sjcl.bitArray.bitSlice(bits, 0, qlen));
+    } else if (qlen < blen) {
+      return sjcl.bn.fromBits(sjcl.bitArray.concat(
+        sjcl.codec.hex.toBits(repeat('0', Math.ceil((qlen - blen) / 4))),
+        curve.r.toBits()
+      ));
+    }
+    return sjcl.bn.fromBits(bits);
+  }
+  function int2octets(integer) {
+    let iModQ = integer.mulmod(new sjcl.bn(1), curve.r);
+
+    let rlen = 8 * Math.ceil(qlen / 8);
+    let ilen = sjcl.bitArray.bitLength(iModQ.toBits());
+
+    return sjcl.bitArray.concat(
+      sjcl.codec.hex.toBits(repeat('0', Math.ceil((rlen - ilen) / 4))),
+      iModQ.toBits()
+    );
+  }
+  function bits2octets(bits) {
+    return int2octets(bits2int(bits).mulmod(new sjcl.bn(1), curve.r));
+  }
+
+  function hmac(bits, key) {
+    let hmacK = new sjcl.misc.hmac(key, hashObject);
```

Comment:
You could make this take the `key` as the first parameter, and then use arguments, and take care of the `sjcl.bitArray.concat` dance which would leave the calling code flatter, and easier to read:

```
k = hmac(
 sjcl.bitArray.concat(
   sjcl.bitArray.concat(v, sjcl.codec.hex.toBits('00')),
   sjcl.bitArray.concat(int2octets(x), bits2octets(hash))
 ),
 k
);
```

becomes 

```
k = hmac(k, 
         v, 
         sjcl.codec.hex.toBits('00'), 
         int2octets(x), 
         bits2octets(hash));
```